stm32: Remove redundant I2C variant Kconfig symbols definitions

erwango opened the pull request:
I2C_STM32_V1 and I2C_STM32_V2 are Kconfig symbols used to
select STM32 I2C driver variant based on I2C IP available on a
given STM32 SoC.
Being SoC dependent, they are automatically selected in soc/
if CONFIG_I2C is set. Then there is no need to define them at board
level, nor for a specific application like this is done in
some sample or test applications. Remove these definitions when
this is the case.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

ulfalizer reviewed (commented):
Could the prompts on `I2C_STM32_V1` and `I2C_STM32_V2` be removed as well? Would it ever make sense for the user to go into the `menuconfig` interface and change those?

erwango committed:
stm32: Remove redundant I2C variant Kconfig symbols definitions

I2C_STM32_V1 and I2C_STM32_V2 are Kconfig symbols used to
select STM32 I2C driver variant based on I2C IP available on a
given STM32 SoC.
Being SoC dependent, they are automatically selected in soc/
if CONFIG_I2C is set. Then there is no need to define them at board
level, nor for a specific application like this is done in
some sample or test applications. Remove these definitions when
this is the case.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
driver/i2c: stm32: Remove prompt on IP version symbols

Kconfig symbols I2C_STM32_V1 and I2C_STM32_V2 depends on SoC reference
and user should not have the possibility to choose one or the other.
Remove prompt on these symbols.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
@ulfalizer, indeed, user shouldn't have any choice here. I've added a commit to remove the prompt. Thanks
